py/vttest/environment.py

```diff
@@ -28,7 +28,14 @@
 
 def get_test_directory():
   """Returns the toplevel directory for the tests. Might create it."""
-  directory = tempfile.mkdtemp(prefix='vt')
+  dataroot = os.environ['VTDATAROOT']
+  if dataroot:
+    directory = os.path.join(dataroot, "test")
```

Comment:
Why didn't I think of that! That just works. I'll add the additional comment on the mysql_flavor & submit.